### PR TITLE
AS-695: convert database to snake case

### PR DESCRIPTION
**Pull request by wheelsandcogs**
* All models should now exclusively use camelCase for property names.
 * All database tables should now exclusively have snake_case column names.

The mapping is performed by `knexSnakeCaseMappers`, for more information see https://vincit.github.io/objection.js/#snake-case-to-camel-case-conversion.

### CODE REVIEW

File: knexfile.js [hunk 1]
```diff
@@ -1,28 +1,36 @@
...
+    port: process.env.DATABASE_PORT,
+    user: process.env.DATABASE_USERNAME || 'postgres',
+    password: process.env.DATABASE_PASSWORD
   }
+};
 
+if (process.env.SNAKE_MAPPER) {
+  test = { ...test, ...knexSnakeCaseMappers() };
+  development = { ...development, ...knexSnakeCaseMappers() };
+}
```
Comment (joefitter) [inline knexfile.js]: Won't this result in the case mappers always being applied? I was thinking more something like:

```js
const mappers = process.env.MIGRATION ? {} : knexSnakeCaseMappers();

module.exports = {
  test: {
    ...mappers,
    [settings]
  },
  development: {
     ...mappers,
    [settings]
  }
}
```

File: index.js [hunk 2]
```diff
@@ -30,7 +30,8 @@ module.exports = connection => {
   const settings = {
     client: 'pg',
     useNullAsDefault: true,
-    connection
+    connection,
+    ...knexSnakeCaseMappers()
```
Comment (joefitter) [inline index.js]: It would be good if we could use the knexfile here, rather than duplicating settings in 2 places

File: knexfile.js (hunk 1 above)
Comment (joefitter) [reply knexfile.js]: actually, yours does exactly the same thing in a slightly more verbose way. I didn't see the last 2 lines 🤦‍♂️

Comment (wheelsandcogs) [reply knexfile.js]: Ah... well in this case you wouldn't set `SNAKE_MAPPER` env var when running the migration command, but I've added it to the npm script for seed?

Comment (wheelsandcogs) [reply knexfile.js]: But I can use your syntax to tidy it up.

File: index.js (hunk 2 above)
Comment (wheelsandcogs) [reply index.js]: Separate ticket? :D

Comment (joefitter) [reply index.js]: nah, it would take longer to write the ticket than to just make the change. But I see what you're saying, stay in your lane 🚗

**Pull request merged**